src/linear_algebra/determinant.lean

```diff
@@ -158,13 +158,13 @@ open_locale classical
 If there is no finite basis on `M`, the result is `1` instead.
 -/
 protected def det : (M →ₗ[A] M) →* A :=
-if H : ∃ (s : set M) (b : basis s A M), s.finite
-then @linear_map.det_aux _ _ _ _ H.some_spec.some_spec.some _ _ _ (trunc.mk H.some_spec.some)
+if H : ∃ (s : finset M), nonempty (basis s A M)
```

Comment:
Is `nonempty (Σ s : finset M, basis s A M)` better or worse in your opinion than `∃ (s : finset M), nonempty (basis s A M)`? (I'm not asking for you to change it, just curious)

Reply:
I don't have a clear reason to prefer one over the other, so in this case I went with the version that looks the most like the previous definition.

Did we end up choosing a spelling for `module.finite_free`?